scipy/cluster/tests/test_hierarchy.py

```diff
@@ -949,7 +949,37 @@ def test_dendrogram_colors(self):
 
         # reset color palette (global list)
         set_link_color_palette(None)
-
+   
+    def test_dendrogram_leaf_colors_zero_dist(self) :
+        # tests that the colors of leafs are correct for tree
+        # with two identical points
+        x = np.array([[1, 0, 0], 
+            [0, 0, 1], 
+            [0, 2, 0], 
+            [0, 0, 1], 
+            [0, 1, 0], 
+            [0, 1, 0]])
+        z = linkage(x, "single")
+        d = dendrogram(z)
```

Comment:
This is the cause of some CI failures.
```suggestion
        d = dendrogram(z, no_plot=True)
```